### PR TITLE
feat: Add Mask from ID Node + ColorField Component Improvements

### PR DESCRIPTION
**Pull request by blessedcoolant**
## Summary

Adds a Mask from ID node that lets you drop in an ID Mask to the node and create a mask based on the color you pick from the ID Map.

The color is currently set to color field (but we need to add an Color Picker) to make it easier to pick the color or can just replace it with hex. Either way works. There's support for both.

https://github.com/invoke-ai/InvokeAI/assets/54517381/4f493386-aa39-4b5b-86ad-c246ff457213

## Updated ColorField Component

- Now also displays the hex code for the color value.
- Fixed up the UI for the Color Picker in the Nodes not filling up the entire space.

https://github.com/invoke-ai/InvokeAI/assets/54517381/04da710f-d651-49cd-b76d-eaebe715e003

## Merge Plan

Can't think of anything to not merge it.

### CODE REVIEW

File: invokeai/frontend/web/src/common/util/colorCodeTransformers.ts
Comment (psychedelicious) [inline invokeai/frontend/web/src/common/util/colorCodeTransformers.ts:1]: Is there any opposition to bringing in a well-tested library to handle the math for us? For example, the packages in https://github.com/colorjs have hundreds of thousands of weekly downloads

Comment (blessedcoolant) [reply invokeai/frontend/web/src/common/util/colorCodeTransformers.ts:1]: None. As long as they have functions that return the data to us exactly as we need and we don't have to do any further operations on top I guess.

Comment (blessedcoolant) [reply invokeai/frontend/web/src/common/util/colorCodeTransformers.ts:1]: Does ColorJS have these functions? I can't seem to find them in their docs.

Comment (psychedelicious) [reply invokeai/frontend/web/src/common/util/colorCodeTransformers.ts:1]: Maybe they don't, nvm then

**Comment by psychedelicious** [comment]
@blessedcoolant Did you see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color that skunkworx found? It has a dropper!

**Comment by blessedcoolant** [comment]
> @blessedcoolant Did you see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color that skunkworx found? It has a dropper!

Yep. Doesn't work.

**Comment by psychedelicious** [comment]
That's unfortunate. On macOS:

- Chrome uses a custom picker with dropper.
- FF uses the OS-provided picker, which also has a dropper.

**Comment by blessedcoolant** [comment]
> That's unfortunate. On macOS:
> 
> * Chrome uses a custom picker with dropper.
> * FF uses the OS-provided picker, which also has a dropper.

Actually it works.  Sorry, there was no feedback from the click so I thought it wasn't working. But clicking outside actually picks a color. However this is only on Chromium browsers on my end.

**Review by psychedelicious — APPROVED** [review]

**Pull request merged**